index.js

```diff
@@ -197,26 +214,90 @@ function uploadMergeFiles(viewId, mergeData, templatesToRecords){
         return Promise.all(uploadPromises);
     })
     .then((uploadResponses) =>{
-        console.log("done uploading everything");
+        log.log("done uploading everything");
+
         if(globalCallback){
             globalCallback(null, "Merge completed successfully");
         }
     })
     .catch(function(err) {
-        handleError(err);
+      checkFieldNames("merged", config.merged_doc_table.view_id);
+      handleError(err);
     });
 }
 
 /**
+ * Function to determine what field is causing the upload error
+ * @param {String} table
+ * @param {Number} viewId
+ */
+function checkFieldNames(table, viewId) {
+  promises = [];
+  promises.push(api.getView(viewId));
+  Promise.all(promises)
+  .then((view) => {
+    if ( promises.length > 0 &&
+        table == "merged" &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_to_template_relationship_field_name) &&
```

Comment:
To make this a lot cleaner I'd suggest first rewriting the config file to put all the field names in one object, then you could just loop over the values in that object and not need to write them all out manually, or add new code when a new config field is added. Something like:

config.js:
```javascript
config.merged_doc_table.fields = {};
....
config.merged_doc_table.fields.merged_document = "File";
config.merged_doc_table.fields.merged_doc_details = "Notes";
```
index.js:
```javascript
var docTableFields = config.merged_doc_table.fields;
for(fieldDesc in docTableFields){
   var fieldName = docTableFields[fieldDesc];
   if(!findDocFieldName(structure, fieldName)){ 
        log.error(`Couldn't find the field \"${fieldName}\" in the table \"${table}\". This value is set in config.js as the value for \"${fieldDesc}\"`);
        return;
   }
```